recipes/rust/all/conanfile.py

```diff
@@ -0,0 +1,89 @@
+from pathlib import PureWindowsPath
+import os.path
+import sys
+
+from conan import ConanFile
+from conan.tools.layout import basic_layout
+from conan.tools.files import copy, get, rename, replace_in_file
+from conan.tools.microsoft import is_msvc
+
+
+class RustConan(ConanFile):
+    name = "rust"
+    description = "The Rust Programming Language"
+    license = "Apache-2.0"
+    topics = ("rust", "language", "rust-language")
+    homepage = "https://www.rust-lang.org"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "PkgConfigDeps", "VirtualBuildEnv"
+    short_paths = True
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def build_requirements(self):
+        self.tool_requires("ninja/1.11.0")
+        self.tool_requires("openssl/3.0.5")
+        self.tool_requires("pkgconf/1.9.3")
+
+    @property
+    def _windows_build_triple(self):
+        suffix = "msvc" if is_msvc(self) else "gnu"
+        return f"{str(self.settings.arch)}-pc-windows-{suffix}"
+
+    def _configure_sources(self):
+        config_file = os.path.join(self.source_folder, "config.toml.example")
+
+        def unix_path(path):
+            if self.settings.os == "Windows":
+                return PureWindowsPath(path).as_posix()
+            return path
+
+        def config(value, replacement):
+            replace_in_file(
+                self,
+                config_file,
+                value,
+                replacement
+            )
+        install_folder = unix_path(os.path.join(self.package_folder, "bin"))
```

Comment:
```suggestion
        install_folder = unix_path(self.package_folder)
```

Would this make the path in the `package_info` command be just `bin` instead of `bin/bin`?

Reply:
Yes. The reason everything is in the bin folder is otherwise it fails the post install hooks (because of the libexec folder).

Reply:
If an exception can be made in the hooks for this package then it would be neater to put everything directl under the package folder.

Reply:
That makes sense. Probably just can't copy the contents of `libexec` into `bin`. :\